### PR TITLE
[WIP] Ignore this, testing 4.11 --> 4.12 upgrades

### PR DESCRIPTION
**Pull request by jkyros**
Trying to stay out of Colin's hair for #3317. 

I think `dn.storeCurrentConfigOnDisk()` before the "special rpm-ostree extra reboot" might be hosing us by making the MCD think it's more done than it actually is when it comes back up. Let's see.

### CODE REVIEW

**Comment by cgwalters** [comment]
Oh yeah, there's an e2e test:

> : [sig-node] nodes should not go unready after being upgraded and go unready only once 

So...I think we just need to get the updated rpm-ostree backported to 4.11 and not try these hacks.

**Comment by jkyros** [comment]
Well that's less fun, but I see what you mean :smile: 

Is there a technical reason we can't live-apply and then immediately rebase out of that to avoid the second reboot? 

LIke, I did: 

```
# deploy from self 
systemd-run --unit machine-config-daemon-update-rpmostree-via-container --collect --wait -- podman run --authfile /var/lib/kubelet/config.json --privileged --pid=host --net=host --rm -v /:/run/host quay.io/jkyros/derived-images:saveme rpm-ostree ex deploy-from-self /run/host

# (back outside the podman container) 
# now make that self deployment live 
rpm-ostree ex apply-live --allow-replacement 

# hmm, but that doesn't seem to change the "system" rpm-ostree, but the new one *is* there on disk, and we can use it as long as we help it find its toys: 
LD_PRELOAD=./sysroot/ostree/deploy/rhcos/var/tmp/ostree-unlock-ovl.KHAPS1/upper/lib64/librhsm.so.0 /sysroot/ostree/deploy/rhcos/var/tmp/ostree-unlock-ovl.KHAPS1/upper/bin/rpm-ostree rebase --experimental ostree-unverified-registry:quay.io/jkyros/derived-images:rhel-coreos-9-18
```

And that does seem to get me to the image I need to be in with only one reboot. 

( What I have done is obviously not particularly elegant -- who would want to fish the binary out of a randomly named directory and LD_PRELOAD its missing lib, but the pieces seem to be mostly there? )

File: pkg/daemon/rpm-ostree.go [hunk 1]
```diff
@@ -194,7 +195,8 @@ func (r *RpmOstreeClient) GetBootedOSImageURL() (string, string, error) {
 			osImageURL = tokens[1]
 		}
 	}
-	return osImageURL, bootedDeployment.Version, nil
+
+	return osImageURL, bootedDeployment.Version, bootedDeployment.Checksum, nil
```
Comment (cgwalters) [inline pkg/daemon/rpm-ostree.go:199]: When client-side changes are involved (layering/overrides), the checksum will differ.  I think what we want here is to use 

```
diff --git a/pkg/daemon/rpm-ostree.go b/pkg/daemon/rpm-ostree.go
index 76432b60b..3cb797394 100644
--- a/pkg/daemon/rpm-ostree.go
+++ b/pkg/daemon/rpm-ostree.go
@@ -38,6 +38,7 @@ type RpmOstreeDeployment struct {
        OSName                  string   `json:"osname"`
        Serial                  int32    `json:"serial"`
        Checksum                string   `json:"checksum"`
+       BaseChecksum            string   `json:"base-checksum,omitempty"`
        Version                 string   `json:"version"`
        Timestamp               uint64   `json:"timestamp"`
        Booted                  bool     `json:"booted"`
```

Then use `BaseChecksum` if it's non-empty.

Comment (jkyros) [reply pkg/daemon/rpm-ostree.go]: Whoops, thanks, I thought I pushed that change up already.

Yeah, I was initially...not thinking about the checksum changing. I figured that out this morning after the `kernelType` e2e test added those kernel RPMs (and of course the checksum changed) and I was like "oh, yeah, duh". :smile:

**Comment by cgwalters** [comment]
[Make verification pass if commit hashes match](https://github.com/openshift/machine-config-operator/pull/3340/commits/ccb1b14e149f683a1fd9aff29f4670ca50f09eb1) makes total sense to me.

**Comment by cgwalters** [comment]
I think we don't need this one anymore!

**Pull request closed without merging**